front/components/assistant/conversation/actions/VisualizationActionIframe.tsx

```diff
@@ -167,6 +182,46 @@ export function VisualizationActionIframe({
     [codeFullyGenerated, iframeLoaded, isErrored]
   );
 
+  const handleVisualizationRetry = async () => {
+    if (retryClicked) {
+      return;
+    }
+    setRetryClicked(true);
+    try {
+      const response = await fetch(
+        `/api/w/${owner.sId}/assistant/conversations/${conversationId}/messages`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            content: `The visualization code failed with this error:\n\`\`\`\n${errorMessage}\n\`\`\`\nPlease fix the code.`,
```

Comment:
Can we store this alongside Vizualisation prompt so it easier to find?

Reply:
The problem is that the file with the viz prompt is backend-only

Reply:
Fair, but then I'd say at least in a variable with `VISUALIZATION` so we avoid magic variable.

Reply:
SG